From 205a3b04e1216c1df199aec95185f7f48b8e0491 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 6 Oct 2015 09:11:38 +0200
Subject: Run only pixbuf-lowmem test on GNU libc

The test case depends on GNU libc specific internals so we only run it
on GNU libc.

This fixes build on Linux with musl libc.

https://bugzilla.gnome.org/show_bug.cgi?id=756078
---
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Backport

 tests/Makefile.am     | 5 -----
 tests/pixbuf-lowmem.c | 8 ++++++++
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index bdd68f3..c122188 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -19,13 +19,8 @@ LDADD = \
 noinst_PROGRAMS += 			\
 	pixbuf-read			\
 	pixbuf-random			\
-	$(NULL)
-
-if OS_LINUX
-noinst_PROGRAMS += 			\
 	pixbuf-lowmem			\
 	$(NULL)
-endif
 
 test_programs = 			\
 	animation			\
diff --git a/tests/pixbuf-lowmem.c b/tests/pixbuf-lowmem.c
index ae71788..9e9571f 100644
--- a/tests/pixbuf-lowmem.c
+++ b/tests/pixbuf-lowmem.c
@@ -24,6 +24,7 @@
 #include <time.h>
 #include <string.h>
 
+#ifdef __GLIBC__
 #define PRETEND_MEM_SIZE (16 * 1024 * 1024)
 #define REMAINING_MEM_SIZE 100000
 
@@ -225,3 +226,10 @@ main (int argc, char **argv)
   
   return 0;
 }
+#else
+int
+main (int argc, char **argv)
+{
+	return 0;
+}
+#endif
-- 
cgit v0.11.2

